test/extended/openstack/loadbalancers.go

```diff
@@ -419,3 +478,56 @@ func createTestDeployment(depName string, labels map[string]string, replicas int
 	}}
 	return testDeployment
 }
+
+// GetFloatingNetworkID returns a floating network ID.
+func GetFloatingNetworkID(client *gophercloud.ServiceClient) (string, error) {
```

Comment:
So this mimics how the cloud provider gets the ID of the floating IP network? What happens if it's forced to be something else in the configuration? IMO test should check the config too. Or you should pass your own network as an annotation when creating the service, but then it's probably not guaranteed the connectivity will always work if you choose something inappropriate for the setup.

Reply:
Exactly, these functions are highly inspired on https://github.com/openshift/cloud-provider-openstack/blob/master/pkg/util/openstack/network.go#L53

The intention of this test is to validate that user can set a previously created FIP on the svc definition, so CCM does not need to create it at all, and it just makes use of it.

Therefore, what is defined on the cloud-provider-config should be irrelevant in this case.

Reply:
@rlobillo: What if there's only one "correct" FIP subnet that can be accessed from where the tests are run and it is set in the config? If autodetection takes incorrect one the connectivity check will fail. I'm just trying to find the corner cases so that when some job with a specific configuration will be run we won't need to debug why this test fails.

Reply:
@dulek: OK! the test will use the subnet  on cloud-provider-config in case it's configured and, if not, auto-detect. I'm on it.